WordPress/Classes/Services/MediaService.m

```diff
@@ -70,27 +75,138 @@ - (void)createMediaWithPHAsset:(PHAsset *)asset
                     completion:(void (^)(Media *media, NSError *error))completion
 {
     NSString *mediaName = [asset originalFilename];
-
-    [self createMediaWith:asset
-              forObjectID:blogObjectID
+    [self exportMediaWith:asset
+                 objectID:blogObjectID
                 mediaName:mediaName
         thumbnailCallback:thumbnailCallback
                completion:completion
      ];
 }
 
-- (void)createMediaWithImage:(nonnull UIImage *)image
-             forBlogObjectID:(nonnull NSManagedObjectID *)blogObjectID
-           thumbnailCallback:(nullable void (^)(NSURL * _Nonnull thumbnailURL))thumbnailCallback
-                  completion:(nullable void (^)(Media * _Nullable media, NSError * _Nullable error))completion
+- (void)createMediaWithImage:(UIImage *)image
+             forBlogObjectID:(NSManagedObjectID *)blogObjectID
+           thumbnailCallback:(void (^)(NSURL *thumbnailURL))thumbnailCallback
+                  completion:(void (^)(Media *media, NSError *error))completion
 {
-    [self createMediaWith:image
-              forObjectID:blogObjectID
+    [self exportMediaWith:image
+                 objectID:blogObjectID
                 mediaName:[[NSUUID UUID] UUIDString]
         thumbnailCallback:thumbnailCallback
                completion:completion];
 }
 
+#pragma mark - Private exporting
+
+- (void)exportMediaWith:(id)exportable
```

Comment:
For this one, the `exportable` object is really just there for the legacy support of using the older method that do in fact conform to the `ExportableAsset` protocol. Also, the `ExportableAsset` protocol is _only_ used in the legacy code and shouldn't be used with any of the new code.

Ideally the `(id)exportable` would be typed specifically to avoid the `isKindOfClass` as seen in the implementation. Maybe create a new protocol in the future when you all remove legacy support?